### PR TITLE
[MIG][12.0] rma_purchase 

### PR DESCRIPTION
**Pull request by murtuzasaleh**
RMA Purchase
===========
This modules extend the RMA functionality allowing to use Purchase Orders as
a RMA source.

### CODE REVIEW

**Comment by murtuzasaleh** [comment]
@lreficent What is the use of this method [def new](https://github.com/Eficent/stock-rma/blame/d9c514009e4439c731846a2bcc934d313d3f944a/rma_purchase/models/purchase_order.py#L11) in rma_purchase module, It is WIP in v11 or other Use of it?
Because it is not called in rma_purchase module.

**Comment by AaronHForgeFlow** [comment]
> @lreficent What is the use of this method [def new](https://github.com/Eficent/stock-rma/blame/d9c514009e4439c731846a2bcc934d313d3f944a/rma_purchase/models/purchase_order.py#L11) in rma_purchase module, It is WIP in v11 or other Use of it?
> Because it is not called in rma_purchase module.

The method new is called when creating a new purchase order. It is like the create method, it is called when creating a new record. That is why is calling super at the beginning.

**Comment by murtuzasaleh** [comment]
@aheficent Can you please review this PR.

**Comment by max3903** [comment]
@osi-scampbell Please review, test and comment. Thanks.

**Comment by murtuzasaleh** [comment]
@osi-scampbell Can you please review this PR.

**Comment by osi-scampbell** [comment]
Code installs clean, smart buttons and fields are added onto an RMA form. No bugs or errors found. Approved

**Comment by max3903** [comment]
@aheficent What are we missing to get this merged?

File: rma_purchase/models/rma_order_line.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        string='Qty To Purchase', copy=False,
+        digits=dp.get_precision('Product Unit of Measure'),
+        readonly=True, compute='_compute_qty_purchase'
+    )
+    qty_purchased = fields.Float(
+        string='Qty Purchased', copy=False,
+        digits=dp.get_precision('Product Unit of Measure'),
+        readonly=True, compute='_compute_qty_purchase'
+    )
+
```
Comment (bjeficent) [inline rma_purchase/models/rma_order_line.py:80]:
```suggestion
    @api.onchange('product_id', 'partner_id')
    def _onchange_product_id(self):
        """Domain for purchase_order_line_id is computed here to make it dynamic."""
        res = super(RmaOrderLine, self)._onchange_product_id()
        if not res.get('domain'):
            res['domain'] = {}
        domain = [
            '|',
            ('order_id.partner_id', '=', self.partner_id.id),
            ('order_id.partner_id', 'child_of', self.partner_id.id)]
        if self.product_id:
            domain.append(('product_id', '=', self.product_id.id))
        res['domain']['purchase_order_line_id'] = domain
        return res
```

**Review by bjeficent — CHANGES_REQUESTED** [review]
I have tested it locally and it works properly. I'll add a change: If a product is selected, add the product to the domain.

**Review by bjeficent — APPROVED** [review]
LGTM :+1:

**Comment by murtuzasaleh** [comment]
@aheficent Can you please review this PR.

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: works fine. Thanks

**Comment by max3903** [comment]
Yay! Thank you!

**Pull request merged**